File: 01_programming_basics/more_exercises/02_conditional_statements/03_harvest.py
import math

square_meters = int(input())
kg_grapes_per_square_meter = float(input())
needed_liters = int(input())
num_workers = int(input())

kg_for_liter = 2.5
total_kg_grapes = square_meters * kg_grapes_per_square_meter
wine = 0.4 * total_kg_grapes / 2.5
difference = abs(wine - needed_liters)
wine_for_workers = difference / num_workers

if wine < needed_liters:
    print(f"It will be a tough winter! More {math.floor(difference)} liters wine needed.")
elif wine >= needed_liters:
    print(f"Good harvest this year! Total wine: {math.floor(wine)} liters.")
    print(f"{math.ceil(difference)} liters left -> {math.ceil(wine_for_workers)} liters per person.")

File: 05_JavaScriptBasics/03_MoreExercises/02_ConditionalStatements/03_harvest.js
function harvest(input){
    let vineyardArea = Number(input[0])
    let grapesPerSquareMeter = Number(input[1])
    let neededLitersWine = Number(input[2])
    let numWorkers = Number(input[3])

    let totalQuantityGrapes = vineyardArea * grapesPerSquareMeter
    let litersWine = totalQuantityGrapes * 0.40 / 2.5

    if (litersWine >= neededLitersWine){
        let leftWine = Math.ceil(litersWine - neededLitersWine)
        let winePerWorker = Math.ceil(leftWine / numWorkers)
        console.log(`Good harvest this year! Total wine: ${Math.floor(litersWine)} liters.`)
        console.log(`${leftWine} liters left -> ${winePerWorker} liters per person.`)
    } else {
        let difference = Math.floor(neededLitersWine - litersWine)
        console.log(`It will be a tough winter! More ${difference} liters wine needed.`)
    }
}
